.github/workflows/smoke-test.yml

```diff
@@ -0,0 +1,132 @@
+            # Pull Request
+            git fetch origin $GITHUB_BASE_REF --depth=1
+            export DIFF=$( git diff --name-only origin/$GITHUB_BASE_REF $GITHUB_SHA )
+            echo "Diff between origin/$GITHUB_BASE_REF and $GITHUB_SHA"
+          else
+            # Push
+            git fetch origin ${{ github.event.before }} --depth=1
+            export DIFF=$( git diff --name-only ${{ github.event.before }} $GITHUB_SHA )
+            echo "Diff between ${{ github.event.before }} and $GITHUB_SHA"
+          fi
+          echo "$DIFF"
+          # Escape newlines (replace \n with %0A)
+          echo "::set-output name=diff::$( echo "$DIFF" | sed ':a;N;$!ba;s/\n/%0A/g' )"
+
+      - name: Set matrix for smoke-tests
+        id: set-matrix
+        run: |
+          DIFF="${{ steps.diff.outputs.diff }}"
+          JSON="{\"directory\":["
+
+          # Loop by lines
+          while read path; do
+            # Set $directory to substring before /
+            directory="$( echo $path | cut -d'/' -f1 -s )"
+
+            if [ "$directory" == .github ]; then
+              continue # Exclude .github directory
+            elif [ "$directory" == deploy ]; then
+              continue # Exclude deploy directory
+            elif [ "$directory" == samples ]; then
+              continue # Exclude samples directory
+            elif [ "$directory" == "" ]; then
+              continue # Exclude empty directory
+            fi
+
+            # Add build to the matrix only if it is not already included
+            JSONline="[\"$directory\"],"
+            if [[ "$JSON" != *"$JSONline"* ]]; then
+              JSON="$JSON$JSONline"
+            fi
+          done <<< "$DIFF"
+
+          # Remove last "," and add closing brackets
+          if [[ $JSON == *, ]]; then
+            JSON="${JSON%?}"
+          fi
+          JSON="$JSON]}"
+
+          # Add different kubernetes versions for the smoke-tests
+          MATRIX_JSON=$(echo "$JSON" | jq --raw-output '.k8s += ["v1.20.2","v1.19.7","v1.18.15","v1.17.17","v1.16.15"]')
+
+          # Set output
+          echo "::set-output name=matrix::$( echo "$MATRIX_JSON" )"
+
+
+  smoke-tests:
+    runs-on: ubuntu-latest
+    name: Run Smoke Tests on "${{ matrix.directory }}" chart
+    needs: generate-matrix
+    strategy:
+      matrix: ${{fromJson(needs.generate-matrix.outputs.matrix)}}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Git Checkout
+      
+      - name: Create Kind Cluster
+        uses: engineerd/setup-kind@v0.5.0
+        with:
+          version: "v0.10.0"
+          config: .github/kind-cluster-${{ matrix.k8s }}.yml
+      
+      - name: Install helm
+        uses: Azure/setup-helm@v1.1
+        with:
+          version: v3.5.3
+      
+      - name: Deploy "${{ matrix.directory }}" chart
+        run: |
+          kubectl create namespace ${{ env.NAMESPACE }}
+          kubectl apply -f deploy/dependencies/mongo.yaml -n ${{ env.NAMESPACE }}
```

Comment:
I'm running https://charts.bitnami.com/bitnami in a single node replicaset internally (not like in my mind). It both allows expansion to a real production mongo deployment and compatibility with our application (transaction). Should offshore the job of maintaining a deployment.

Reply:
Relevant bits from my values

`
clusterDomain: cluster.local
architecture: replicaset
useStatefulSet: true

auth:
  enabled: true
  rootPassword: "<thou shalt not>"
  replicaSetKey: ""
replicaSetName: rs0
replicaSetHostnames: true
replicaCount: 1

persistence:
  enabled: false

service:
  nameOverride: ""
  type: ClusterIP
  port: 27017
  portName: mongodb
  nodePort: ""
  externalIPs: []
  loadBalancerSourceRanges: []
  annotations: {}

externalAccess:
  enabled: true
  service:
    type: LoadBalancer
    port: 27017
    loadBalancerIPs: 
      - {{ metallb_ip_ini }}
    domain: ara.dev

arbiter:
  enabled: true
`

Reply:
Yeah, I dunno how to format yaml in a comment.

Reply:
i don't think we need the bitnami charts as it uses StatefulSets + a bunch of other stuff. A simple deployments should suffice. However, we have redis/mongodb charts in the cloudsmith that could be used.